runtime/drivers/bigquery/sql_store.go

```diff
@@ -0,0 +1,210 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+	"strings"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/civil"
+	"github.com/rilldata/rill/runtime/drivers"
+	"go.uber.org/zap"
+)
+
+// Exec implements drivers.SQLStore
+func (c *Connection) Exec(ctx context.Context, src *drivers.DatabaseSource) (drivers.RowIterator, error) {
+	props, err := parseSourceProperties(src.Props)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := c.createClient(ctx, props)
+	if err != nil {
+		if strings.Contains(err.Error(), "unable to detect projectID") {
+			return nil, fmt.Errorf("projectID not detected in credentials. Please set project ID")
+		}
+		return nil, fmt.Errorf("failed to create bigquery client: %w", err)
+	}
+
+	if err := client.EnableStorageReadClient(ctx); err != nil {
+		client.Close()
+		return nil, err
+	}
+
+	q := client.Query(src.Query)
+	it, err := q.Read(ctx)
+	if err != nil && !strings.Contains(err.Error(), "Syntax error") {
+		c.logger.Info("query failed, retyring without storage api", zap.Error(err))
+		// the query results are always cached in a temporary table that storage api can use
+		// there are some exceptions when results aren't cached
+		// so we also try without storage api
+		client, err = c.createClient(ctx, props)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create bigquery client: %w", err)
+		}
+
+		q := client.Query(src.Query)
+		it, err = q.Read(ctx)
+	}
+	if err != nil {
+		client.Close()
+		return nil, err
+	}
+
+	return &rowIterator{
+		client: client,
+		bqIter: it,
+	}, nil
+}
+
+type rowIterator struct {
+	client  *bigquery.Client
+	next    []any
+	nexterr error
+	schema  drivers.Schema
+	bqIter  *bigquery.RowIterator
+}
+
+var _ drivers.RowIterator = &rowIterator{}
+
+func (r *rowIterator) ResultSchema(ctx context.Context) (drivers.Schema, error) {
+	if r.schema != nil {
+		return r.schema, nil
+	}
+
+	// schema is only available after first next call
+	r.next, r.nexterr = r.Next(ctx)
+	if r.nexterr != nil {
+		return nil, r.nexterr
+	}
+
+	r.schema = make([]drivers.Field, len(r.bqIter.Schema))
+	for i, s := range r.bqIter.Schema {
+		dbt, err := bqToDuckDB(string(s.Type))
+		if err != nil {
+			return nil, err
+		}
+
+		r.schema[i] = drivers.Field{Name: s.Name, Type: dbt}
+	}
```

Comment:
1. I don't think we have such a util yet
2. If it's needed, you can just add it. But I'm not sure it's needed. The SQL types around timestamp time zones is not what you would expect, it's not as simple as just "with or without" timezone. For example, in DuckDB, `TIMESTAMP` is not "without" timezone, rather it is always in UTC.

Reply:
2. Yeah internally it may still be normalized to UTC but we would need it to set correct datatype for duckdb column which user may use in their models.